cin_validator/rules/cin2022_23/rule_8696.py

```diff
@@ -0,0 +1,95 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import (
+    CINTable,
+    IssueLocator,
+    RuleContext,
+    rule_definition,
+)
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+# Replace ChildIdentifiers with the table name, and LAChildID with the column name you want.
+
+Assessments = CINTable.Assessments
+Header = CINTable.Header
+AssessmentAuthorisationDate = Assessments.AssessmentAuthorisationDate
+ReferenceDate = Header.ReferenceDate
+
+# define characteristics of rule
+@rule_definition(
+    code=8696,
+    module=CINTable.Assessments,
+    message="Assessment end date must fall within the census year",
+    affected_fields=[AssessmentAuthorisationDate, ReferenceDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[Assessments]
+    df_ref = data_container[Header]
+
+    ref_data_series = df_ref[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_data_series)
+
+    # implement rule logic as described by the Github issue. Put the description as a comment above the implementation as shown.
+
+    # <If present <AssessmentAuthorisationDate> (N00160) must be on or between [Start_Of_Census_Year] and <ReferenceDate> (N00603)
+
+    df = df[
+        ~(
+            (df[AssessmentAuthorisationDate] < collection_start)
+            & (df[AssessmentAuthorisationDate] > collection_end)
+        )
+    ]
```

Comment:
We are almost there.
Let's look at this again.
We want to filter the values that are not in the period of census. 

However, the code logic in these lines reads as follows:
- line 45 selects values that are before the census start (good)
- line 46 selects values that are after the census end (good)
- However, a single value being examined cannot be before and after a period. So instead of using "and" (that is, &) to combine the boolean conditions in line 45 and 46, use "or".

If that is done, there is no need to negate the filter by putting ~ on the outer bracket in line 44. What that does is say that "do the opposite of whatever the inner bracket (line 45 and 46) have done".

If you combine the conditions on line 45 and 46 with an "or" comparator (|) , your boolean mask is complete already.

Reply:
Hi @tab1tha 

I've amended the code and it now appears to work without any errors. I think I've now sent it back to you for review.

Any problems, let me know.

Thanks,
Gavin